libmproxy/protocol/base.py

```diff
@@ -109,9 +109,9 @@ def __call__(self):
                         self.disconnect()
     """
 
-    def __init__(self, server_address=None):
+    def __init__(self, server_address=None, source_address=None):
         super(ServerConnectionMixin, self).__init__()
-        self.server_conn = ServerConnection(server_address)
+        self.server_conn = ServerConnection(server_address, source_address)
```

Comment:
Can we take this out of the constructor here and just use the config value?
(This would also remove the other proxy mode cases)